syncer doesn't rotate to the next binlog file when no RotateEvent in the current binlog file

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

   stop and start the upstream MySQL server when replicating incremental data.

2. What did you expect to see?

    the task resumed automatically after the upstream MySQL restarted.

3. What did you see instead?

    ```
    [2020/09/15 15:17:56.076 +08:00] [ERROR] [subtask.go:295] ["unit process error"] [subtask=task_single] [unit=Sync] ["error information"="{\"ErrCode\":36001,\"ErrClass\":\"sync-unit\",\"ErrScope\":\"internal\",\"ErrLevel\":\"high\",\"Message\":\"panic error: table checkpoint position: (mysql-bin.000002, 568), gtid-set: 074be7f4-f0f1-11ea-95bd-0242ac120002:1-153 less than global checkpoint position: (mysql-bin.000002, 25660968), gtid-set: 074be7f4-f0f1-11ea-95bd-0242ac120002:1-153(flushed position: (mysql-bin.000002, 25660968), gtid-set: 074be7f4-f0f1-11ea-95bd-0242ac120002:1-153)\"}"]
    ```

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        v2.0.0-rc.2
        ```
[dm-worker-checkpoint.log](https://github.com/pingcap/dm/files/5223473/dm-worker-checkpoint.log)

--- a/syncer/syncer.go
+++ b/syncer/syncer.go
@@ -1499,7 +1499,7 @@ type eventContext struct {
 func (s *Syncer) handleRotateEvent(ev *replication.RotateEvent, ec eventContext) error {
 	if ec.header.Timestamp == 0 || ec.header.LogPos == 0 { // fake rotate event
 		if string(ev.NextLogName) <= ec.lastLocation.Position.Name {
-			return nil // not rotate to the next binlgo file, ignore it
+			return nil // not rotate to the next binlog file, ignore it
 		}
 	}
 

--- a/syncer/syncer.go
+++ b/syncer/syncer.go
@@ -1497,9 +1497,10 @@ type eventContext struct {
 // TODO: Further split into smaller functions and group common arguments into
 // a context struct.
 func (s *Syncer) handleRotateEvent(ev *replication.RotateEvent, ec eventContext) error {
-	if ec.header.Timestamp == 0 || ec.header.LogPos == 0 {
-		// it is fake rotate event, ignore it
-		return nil
+	if ec.header.Timestamp == 0 || ec.header.LogPos == 0 { // fake rotate event
+		if string(ev.NextLogName) <= ec.lastLocation.Position.Name {
+			return nil // not rotate to the next binlgo file, ignore it
+		}
 	}
 
 	*ec.currentLocation = binlog.Location{

